### DISCUSSION TITLE
Extracting Specific Features

### DISCUSSION THREAD

**Discussion by TheFilS** [blue-yonder/tsfresh#1064] [General]
I'm having difficulty extracting specific features. So, I run TSFresh and extract all features and train a decision tree. From the decision tree I take the features used in classification. Then I want to run TSFresh to extract features from a new set of data so that I can run it through the decision tree. I don't want to extract all features again, but only from the list that my decision tree ended up using.

For example running:
my_list = top_feature_list
my_dict = {}
for index in my_list:
    my_dict[index] = "none"

print(my_dict)

I get --
{'value__partial_autocorrelation__lag_4': 'none', 'value__ratio_beyond_r_sigma__r_7': 'none', 'value__number_peaks__n_5': 'none', 'value__agg_autocorrelation__f_agg_"var"__maxlag_40': 'none', 'value__count_below__t_0': 'none', 'value__lempel_ziv_complexity__bins_3': 'none', 'value__change_quantiles__f_agg_"mean"__isabs_True__qh_0.8__ql_0.4': 'none', 'value__fft_aggregated__aggtype_"centroid"': 'none'}

When I try to re extract using:
X = tsfresh.extract_features(df, column_id="id", column_value='value', default_fc_parameters=my_dict)

I get the following error message:
AttributeError: module 'tsfresh.feature_extraction.feature_calculators' has no attribute 'value__partial_autocorrelation__lag_4'

I'm guessing the value__partial_autocorrelation is the actual attribute and that "lag_4" needs to somehow be specifies as a daughter sub-class of the feature but I have no idea how to do this. I couldn't really find anything. 

Any help would be great thanks so much!

**Comment by nils-braun** [comment]
Hi @TheFilS - sorry for the late answer. The use-case you described is exactly what the `from_columns` is for: https://tsfresh.readthedocs.io/en/stable/api/tsfresh.feature_extraction.html#tsfresh.feature_extraction.settings.from_columns. It will return a settings object that you can use for the feature extraction.
You can also compare with a built-in extraction settings object. They are all just dictionaries :)
